src/openmmio.py

```diff
@@ -888,7 +919,27 @@ def update_simulation(self, **kwargs):
         else:
             self.create_simulation(**self.simkwargs)
 
-    def set_positions(self, shot=0, traj=None):
+    def set_restraint_positions(self, shot):
+        """
+        Set reference positions for energy restraints.  This may be a different set of positions
+        from the "current" positions that are stored in self.mol and self.xyz_omm.
+        """
+        if hasattr(self, 'restrain_k'):
+            ## Generate OpenMM-compatible positions in nanometers
+            xyz = self.ref_mol.xyzs[shot] / 10.0
+            have_restraint_frc = False
+            for frc in self.simulation.system.getForces():
+                if isinstance(frc, CustomExternalForce):
+                    have_restraint_frc = True
+                    for i, j in enumerate(self.realAtomIdxs):
+                        frc.setParticleParameters(i, j, [xyz[i, 0], xyz[i, 1], xyz[i, 2]])
```

Comment:
This is a small suggestion: I wonder if `frc.setParticleParameters(i, j, xyz[i])` works here, if not we can also consider replacing line 929 with
```
xyz = (self.ref_mol.xyzs[shot] / 10.0).tolist()
```

Reply:
Changes have been made in the latest commit.